gateway/src/endpoints/inference.rs

```diff
@@ -84,6 +87,20 @@ struct InferenceMetadata<'a> {
     pub previous_model_inference_results: Vec<ModelInferenceResponseWithMetadata<'a>>,
 }
 
+#[derive(Debug, Deserialize, Default)]
+pub struct InferenceApiKeys {
+    pub anthropic_api_key: Option<SecretString>,
+    pub azure_openai_api_key: Option<SecretString>,
+    pub fireworks_api_key: Option<SecretString>,
+    pub gcp_vertex_api_key: Option<SecretString>,
+    pub mistral_api_key: Option<SecretString>,
+    pub openai_api_key: Option<SecretString>,
+    pub together_api_key: Option<SecretString>,
+    pub vllm_api_key: Option<SecretString>,
+    #[cfg(any(test, feature = "e2e_tests"))]
+    pub dummy_api_key: Option<SecretString>,
+}
```

Comment:
Double nest it: e.g. `openai.api_key` and so on.

Skip azure for now since it depends on deployment ID and other factors (to be implemented later.

This allows us to customize more parameters later (e.g. different endpoints and so on).